### PR TITLE
Restrict discouragement of ArrayObject to wrapping objects

### PR DESCRIPTION
**Pull request by cmb69**
This is a follow-up to PR #3858, based on a comment by @arnaud-lb[1].

[1] <https://github.com/php/php-src/issues/16371#issuecomment-2410928030>

### CODE REVIEW

**Review by arnaud-lb — APPROVED** [review]
Thank you @cmb69! The change looks good to me. I think we could change the class description as well, to reflect the intended use-case more.

**Review by Girgias — APPROVED** [review]

**Review by jimwins — APPROVED** [review]
LGTM with one word change.

File: reference/spl/arrayobject.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
    </para>
    <note>
     <simpara>
-     This class is conceptionally flawed, and therefore its usage is discouraged.
+     Wrapping objects with this class is conceptionally flawed, and therefore its usage with objects is discouraged.
```
[the remark this answers is not in the document]
Comment (cmb69) [reply reference/spl/arrayobject.xml]: Thank you! And yeah, I like "fundamentally" even more.

**Comment by cmb69** [comment]
@arnaud-lb, it seems that more should be changed than the class description. E.g. https://www.php.net/manual/en/arrayobject.count.php claims:

> Get the number of public properties in the ArrayObject

The whole documentation of this class should be revised. I'll try to come up with a more complete PR soonish, but I'm going to merge this now so users are not discouraged to wrap arrays (an interesting use-case I've never considered before).

**Pull request merged**